### PR TITLE
feat: Migrate getAllActive -> spenderPlatformV1ApiService

### PR DESCRIPTION
**Pull request by SahilK-027**
## Clickup
[Please add link here](https://app.clickup.com/t/86cv9n9gt)

### CODE REVIEW

**Review by Dimple16 — APPROVED** [review]

File: src/app/core/services/projects.service.spec.ts [hunk 1]
```diff
@@ -83,17 +86,18 @@ describe('ProjectsService', () => {
   });
 
   it('should be able to fetch all active projects', (done) => {
-    apiService.get.and.returnValue(of(apiResponseActiveOnly));
+    spenderPlatformV1ApiService.get.and.returnValue(of(platformAPIResponseActiveOnly));
+    const transformToV1ResponseSpy = spyOn(projectsService, 'transformToV1Response').and.callThrough();
```
Comment (devendrafyle) [inline src/app/core/services/projects.service.spec.ts:90]: Could you please check how we are testing the `transform` method in platform expense APIs, in that case, we are testing it from the `transaction` service the only change in your case is that you have to test it from the project service itself. 

By testing that way we not only ensure that the `transform` method is called, but also we test for what input what output to expect.

**Review by devendrafyle — APPROVED** [review]
minor comment

File: src/app/core/services/projects.service.spec.ts (hunk 1 above)
Comment (suyashpatil78) [reply src/app/core/services/projects.service.spec.ts:90]: +1

File: src/app/core/services/projects.service.spec.ts [hunk 2]
```diff
@@ -83,17 +86,18 @@ describe('ProjectsService', () => {
...
     projectsService.getAllActive().subscribe((res) => {
       expect(res).toEqual(expectedReponseActiveOnly);
+      expect(spenderPlatformV1ApiService.get).toHaveBeenCalledWith('/projects', {
```
Comment (suyashpatil78) [inline src/app/core/services/projects.service.spec.ts:93]: `toHaveBeenCalledOnceWith`

**Review by suyashpatil78 — COMMENTED** [review]
2 comments

File: src/app/core/services/projects.service.spec.ts (hunk 1 above)
Comment (SahilK-027) [reply src/app/core/services/projects.service.spec.ts:90]: OK

File: src/app/core/services/projects.service.spec.ts (hunk 2 above)
Comment (SahilK-027) [reply src/app/core/services/projects.service.spec.ts:93]: Yup I will change it 👍

File: src/app/core/services/projects.service.spec.ts (hunk 1 above)
Comment (SahilK-027) [reply src/app/core/services/projects.service.spec.ts:90]: Will work on this in separate PR for all the methods at once.

cc: @devendrafyle , @suyashpatil78

**Review by suyashpatil78 — APPROVED** [review]

**Pull request merged**